Remove broken unused constructor. Fixes build error with clang 11:

error: no viable overloaded '='

https://bugs.scribus.net/view.php?id=16605
https://github.com/scribusproject/scribus/commit/34dee8142844a7ee70ddff95253d0299896bcc16
https://gitlab.com/inkscape/lib2geom/-/commit/014438214a837e8193edd342c9f701b1858e493c#ab2e5db20b80be8d13059370abeebc3ca83bff73_178_250
--- scribus/plugins/tools/2geomtools/lib2geom/path.h.orig
+++ scribus/plugins/tools/2geomtools/lib2geom/path.h
@@ -250,15 +250,6 @@
   double valueAt(double t, Dim2 d) const { return inner[d].valueAt(t); }
 
   D2<SBasis> toSBasis() const {return inner.toSBasis(); }
-
-protected:
-  BezierCurve(Point c[]) {
-    Coord x[order+1], y[order+1];
-    for(unsigned i = 0; i <= order; i++) {
-        x[i] = c[i][X]; y[i] = c[i][Y];
-    }
-    inner = Bezier(x, y);
-  }
 };
 
 // BezierCurve<0> is meaningless; specialize it out
